sweep/store.go

```diff
@@ -0,0 +1,234 @@
+package sweep
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/coreos/bbolt"
+	"github.com/lightningnetwork/lnd/channeldb"
+)
+
+var (
+	// lastTxBucketKey is the key that points to a bucket containing a
+	// single item storing the last published tx.
+	//
+	// maps: lastTxKey -> serialized_tx
+	lastTxBucketKey = []byte("sweeper-last-tx")
+
+	// lastTxKey is the fixed key under which the serialized tx is stored.
+	lastTxKey = []byte("last-tx")
+
+	// txHashesBucketKey is the key that points to a bucket containing the
+	// hashes of all sweep txes that were published successfully.
+	//
+	// maps: txHash -> empty slice
+	txHashesBucketKey = []byte("sweeper-tx-hashes")
+
+	// utxnChainPrefix is the bucket prefix for nursery buckets.
```

Comment:
A pretty big information leak for the sweeper to be aware of the individual keys that the bucket structure of the nursery itself. I understand this is mean to be a one time thing, but it's a bit fragile. Instead, we can introduce an interface to fetch the information we need from the nursery. This was we can easily unit test it by mocking out this new interface, and we don't need to burden the sweeper itself with knowledge of the database structure of the nursery.

Reply:
If this would have been a db migration, we'd have the same thing. The idea is that even after the nursery is long gone, people upgrading from an old version are still able to recover this data.